L1Trigger/L1THGCal/src/backend_emulator/HGCalHistoSeeding_SA.cc

```diff
@@ -212,21 +218,27 @@ void HGCalHistoSeeding::calculateAveragePosition(HGCalHistogramCellSAPtrCollecti
   const unsigned int stepLatency = config_.getStepLatency(CalcAverage);
   for (auto& hc : histogram) {
     hc->addLatency(stepLatency);
+    const unsigned nBits = 17;
     if (hc->N() > 0) {
-      unsigned int inv_N = int(round(1.0 * 0x1FFFF / hc->N()));
-      hc->setX((hc->X() * inv_N) >> 17);
-      hc->setY((hc->Y() * inv_N) >> 17);
+      unsigned int inv_N = int(round(float(0x1 << nBits) / hc->N()));
+      hc->setX((hc->X() * inv_N) >> nBits);
+      hc->setY((hc->Y() * inv_N) >> nBits);
     }
   }
 }
 
 // Useful debugging function for printing histogram contents
 void HGCalHistoSeeding::printHistogram(const HGCalHistogramCellSAPtrCollection& histogram) const {
+  std::string histogramString = "";
   for (unsigned int iRow = 0; iRow < config_.cRows(); ++iRow) {
     for (unsigned int iCol = 0; iCol < config_.cColumns(); ++iCol) {
       unsigned binIndex = config_.cColumns() * iRow + iCol;
-      std::cout << histogram.at(binIndex)->S() << " ";
+      // std::cout << histogram.at(binIndex)->S() << " ";
+      histogramString += histogram.at(binIndex)->S();
+      histogramString += " ";
     }
-    std::cout << std::endl;
+    histogramString += "\n";
   }
+  edm::LogInfo("HGCalHistoSeeding") << "Histogram :";
+  edm::LogInfo("HGCalHistoSeeding") << histogramString;
```

Comment:
Since MessageLogger won't be available in the standalone emulator, this should be outside of the emulator code

Reply:
Removed function.